src/components/FormComponents/Pasientopplysninger/Adresser/Oppholdsadresse.tsx

```diff
@@ -0,0 +1,53 @@
+import { BodyShort, Heading } from '@navikt/ds-react';
+
+import { Oppholdsadresse } from '../../../../graphql/queries/graphql.generated';
+
+import Vegadresse from './Vegadresse';
+import Matrikkeladresse from './Matrikkeladresse';
+import UtenlandskAdresse from './Utenlandskadresse';
+
+type OppholdsadresseProps = {
+    adresser?: Oppholdsadresse | null;
+};
+
+type OppholdAnnetType = 'MILITAER' | 'PENDLER' | 'UTENRIKS' | 'PAA_SVALBARD' | string;
+
+function oppholdAnnet(value: OppholdAnnetType): string {
+    switch (value) {
+        case 'MILITAER':
+            return 'Militær';
+        case 'PENDLER':
+            return 'Pendler';
+        case 'UTENRIKS':
+            return 'Utenriks';
+        case 'PAA_SVALBARD':
+            return 'På Svalbard';
+        default:
+            return value;
+    }
+}
+
+function Oppholdsadresse({ adresser }: OppholdsadresseProps): JSX.Element | null {
+    if (!adresser) return null;
+
+    return (
+        <div>
+            <Heading level="3" size="xsmall">
+                Oppholdsadresse
+            </Heading>
+            {adresser.vegadresse ? (
+                <Vegadresse vegadresse={adresser.vegadresse} />
+            ) : adresser.matrikkeladresse ? (
+                <Matrikkeladresse matrikkeladresse={adresser.matrikkeladresse} />
+            ) : adresser.utenlandskAdresse ? (
+                <UtenlandskAdresse utenlandskAdresse={adresser.utenlandskAdresse} />
+            ) : adresser.oppholdAnnetSted ? (
+                <BodyShort>{`Annet opphold: ${oppholdAnnet(adresser?.oppholdAnnetSted)}`}</BodyShort>
+            ) : (
+                <BodyShort>Oppholdsadresse mangler</BodyShort>
+            )}
+        </div>
+    );
+}
```

Comment:
Er ikke så fan av å chaine så mange ternaries, litt mer eksplisitt, men hva med f.eks:

```suggestion
function Oppholdsadresse({ adresser }: OppholdsadresseProps): JSX.Element | null {
    if (!adresser) return null;

    return (
        <div>
            <Heading level="3" size="xsmall">
                Oppholdsadresse
            </Heading>
            <FirstAvailableAdress adresse={adresser} />
        </div>
    );
}

function FirstAvailableAdress({ adresser}: OppholdsadresseProps): JSX.Element {
    if (adresser.vegadresse) {
        return <Vegadresse vegadresse={adresser.vegadresse} />
    } else if (adresser.matrikkeladresse) {
        return <Matrikkeladresse matrikkeladresse={adresser.matrikkeladresse} />
    } else if (adresser.utenlandskAdresse) {
        return <UtenlandskAdresse utenlandskAdresse={adresser.utenlandskAdresse} />
    } else if (adresser.oppholdAnnetSted) {
        return <BodyShort>{`Annet opphold: ${oppholdAnnet(adresser?.oppholdAnnetSted)}`}</BodyShort>
    } else {
        return <BodyShort>Oppholdsadresse mangler</BodyShort>
    }
}
```

eller noe i den duren?